current_spring2026/ingestion/ingest.py

```
@@ -0,0 +1,394 @@
+
+# ── DB writes ─────────────────────────────────────────────────────────────────
+
+def upsert_documents(
+    records: List[dict],
+    meta_embeddings: np.ndarray,
+    meta_sparse: List[dict],
+    conn,
+) -> dict:
+    """Upsert into documents table. Returns {ark_id: db_id}."""
+    rows = []
+    for rec, emb, sparse in zip(records, meta_embeddings, meta_sparse):
```

Ruff (0.15.9)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)